barretenberg/cpp/src/barretenberg/transcript/transcript.hpp

```diff
@@ -109,83 +113,82 @@ class BaseTranscript {
-        previous_challenge_buffer = new_challenge_buffer;
-        return new_challenge_buffer;
+        previous_challenge = new_challenge;
+        return new_challenge;
     };
 
   protected:
     /**
      * @brief Adds challenge elements to the current_round_buffer and updates the manifest.
      *
      * @param label of the element sent
-     * @param element_bytes serialized
+     * @param element_frs serialized
      */
-    void consume_prover_element_bytes(const std::string& label, std::span<const uint8_t> element_bytes)
+    void consume_prover_element_frs(const std::string& label, std::span<const Fr> element_frs)
     {
         // Add an entry to the current round of the manifest
-        manifest.add_entry(round_number, label, element_bytes.size());
+        manifest.add_entry(round_number, label, element_frs.size());
 
-        current_round_data.insert(current_round_data.end(), element_bytes.begin(), element_bytes.end());
+        current_round_data.insert(current_round_data.end(), element_frs.begin(), element_frs.end());
 
-        num_bytes_written += element_bytes.size();
+        num_frs_written += element_frs.size();
     }
 
     /**
      * @brief Serializes object and appends it to proof_data
      * @details Calls to_buffer on element to serialize, and modifies proof_data object by appending the serialized
-     * bytes to it.
+     * frs to it.
      * @tparam T
      * @param element
      * @param proof_data
      */
     template <typename T> void serialize_to_buffer(const T& element, Proof& proof_data)
     {
-        auto element_bytes = to_buffer(element);
-        proof_data.insert(proof_data.end(), element_bytes.begin(), element_bytes.end());
+        auto element_frs = bb::field_conversion_utils::convert_to_bn254_frs(element);
+        proof_data.insert(proof_data.end(), element_frs.begin(), element_frs.end());
     }
     /**
-     * @brief Deserializes the bytes starting at offset into the typed element and returns that element.
-     * @details Using the template parameter and the offset argument, this function deserializes the bytes with
-     * from_buffer and then increments the offset appropriately based on the number of bytes that were deserialized.
+     * @brief Deserializes the frs starting at offset into the typed element and returns that element.
+     * @details Using the template parameter and the offset argument, this function deserializes the frs with
+     * from_buffer and then increments the offset appropriately based on the number of frs that were deserialized.
      * @tparam T
      * @param proof_data
      * @param offset
      * @return T
      */
     template <typename T> T deserialize_from_buffer(const Proof& proof_data, size_t& offset) const
     {
-        constexpr size_t element_size = sizeof(T);
-        ASSERT(offset + element_size <= proof_data.size());
+        constexpr size_t element_fr_size = bb::field_conversion_utils::calc_num_frs<T>();
```

Comment:
could maybe merge this function call with `convert_from_bn254_frs` in a better way